packages/rps/src/utils/channelClient.ts

```diff
@@ -0,0 +1,53 @@
+import {BigNumberish} from 'ethers/utils';
+import EventEmitter from 'eventemitter3';
+import {GameState} from '../redux/game/state';
+
+export enum ChannelState {
+  Proposed = 'proposed',
+  Open = 'open',
+}
+
+export type MessageParameters = OpenChannelParameters | JoinChannelParameters;
+
+export interface OpenChannelParameters {
+  participants: string[];
+  balances: BigNumberish[];
+  appData: GameState;
+  state: 'proposed';
+}
+
+export interface JoinChannelParameters {
+  participants: string[];
+  channelId: string[];
+  state: 'open';
+}
+
+export class ChannelClient {
+  protected events: EventEmitter = new EventEmitter();
+
+  constructor() {
+    this.events.on('channelUpdated', (payload: MessageParameters) => {
+      this.sendMessage('chan_update', payload);

```

Comment:
I think you probably don't quite want this, as `channelUpdated` can be triggered in one of two ways: (1) when the app calls a method like `updateState`, (2) when a `channelUpdated` event arrives via `messageReceived`. You only want to `sendMessage` in the first of these situations, as otherwise you will end up in an infinite loop!

You probably want to explicitly trigger a `sendMessage` in the `openChannel`, `joinChannel` and `updateState` methods.